secrets/certregistry/certregistry.go

```diff
@@ -85,12 +99,125 @@ func (r *CertRegistry) SyncCert(key string, hosts []string, crt *tls.Certificate
 // GetCertFromHello reads the SNI from a TLS client and returns the appropriate certificate.
 // If no certificate is found for the host it will return a default certificate.
 func (r *CertRegistry) GetCertFromHello(hello *tls.ClientHelloInfo) (*tls.Certificate, error) {
+	var certList = make([]*tlsCertificate, 0)
 	for _, cert := range r.lookup {
 		for _, host := range cert.hosts {
 			if hello.ServerName == host {
-				return cert.cert, nil
+				certList = append(certList, cert)
 			}
 		}
 	}
-	return r.defaultTLSCert, nil
+
+	switch len(certList) {
+	case 0:
+		// no certs found so returning the default certificate
+		return r.defaultTLSCert, nil
+	case 1:
+		// only one certificate found, retuning it
+		return certList[0].cert, nil
+	default:
+		// more that one cert found, returning the best match
+		log.Warnf("found more than one certificate for %s, returning best one", hello.ServerName)
```

Comment:
no warning, it's expected to have it and this would flood on every connection the logs :)

Reply:
The reason I added the warning is this would only ever execute if there was two different ingresses using different secrets terminating the same exact host. IMO this is most likely a configuration mistake since it would be almost impossible to determine which Ingress to use.

Reply:
Maybe to add: we have something that generates up to hundreds of ingress from another CRD, so I think it's fine to serve multiple ingress with the same host and while rotating certs you will have a split, some ingress might be updated already and others not yet, but skipper already got all ingress configuring the routes and TLS certs/secrets...

Reply:
Thanks for the clarification! That actually makes a lot of sense.